### PR TITLE
make FastifySchemaValidationError.params wider

### PR DESCRIPTION
**Pull request by cm-ayf**
[`FastifySchemaValidationError`](https://github.com/learnwiz/fastify/blob/4bde0e0b259b8df6495ae93c91e06615b61be9c6/types/schema.d.ts#L26) seems to originate from [`ErrorObject` from Ajv](https://github.com/ajv-validator/ajv/blob/main/lib/types/index.ts#L85).  
However, while property `params` in `FastifySchemaValidationError` has type `Record<string | string[]>`, possible values that are created in Ajv are wider than this; see [`LimitNumberError`](https://github.com/ajv-validator/ajv/blob/main/lib/vocabularies/validation/limitNumber.ts#L18-L22) and [`ConstError`](https://github.com/ajv-validator/ajv/blob/main/lib/vocabularies/validation/const.ts#L7). Note that `allowedValue` defined in `ConstError` could be literally any; see [JSON Schema Validation Specification](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-const).  
Property `params` in `FastifySchemaValidationError` should be wider as `Record<string, unknown>` (or `Record<string, any>` for better coding experience?) to accept those values.

By the way, there were duplicated definition (`ValidationResult`) for this, so I refactored them here.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by cm-ayf** [comment]
I noticed that `ValidationResult` was exported from fastify.d.ts.  
I temporarily added alias for backword compatibility, but would follow recommendation.

File: test/types/fastify.test-d.ts [hunk 1]
```diff
@@ -234,7 +233,6 @@ const ajvErrorObject: AjvErrorObject = {
   params: {},
   message: ''
 }
-expectAssignable<ValidationResult>(ajvErrorObject)
```
Comment (Eomm) [inline test/types/fastify.test-d.ts:237]: I would add a guard here to run the assertion when the `params` is a string

**Review by mcollina — APPROVED** [review]
lgtm

File: test/types/fastify.test-d.ts (hunk 1 above)
Comment (cm-ayf) [reply test/types/fastify.test-d.ts:237]: I'm sorry for late response. Did you mean to have a test like implemented in 9f6008f0353bfa9ea89e9f1e4334060a427574f2?

**Comment by mcollina** [comment]
I'm sorry this slipped and it's pretty old. Could you rebase so we can land?

**Comment by cm-ayf** [comment]
Sorry for late response, but I've just rebased the branch.
The test failed, but it seems that’s not from my change (in `test/pretty-print.test.js`), so I'd just like to merge this PR.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**